### PR TITLE
Simplify GQA conditions in sdpa_attention.py

### PR DESCRIPTION
**Pull request by justinchuby**
Removed unnecessary checks for key being a torch.fx.Proxy in GQA conditions because fx tracing is no longer supported, and torch.export supports enable_gqa.

### CODE REVIEW

**Review by vasqu — APPROVED** [review]
Perfect, thank you! Makes more sense esp since we removed fx support explicitly in #41683

File: src/transformers/integrations/sdpa_attention.py [hunk 1]
```diff
@@ -32,13 +32,11 @@ def use_gqa_in_sdpa(attention_mask: Optional[torch.Tensor], key: torch.Tensor) -
     # 1.cuda or Ascend NPU
     #   - torch version >= 2.5
     #   - attention_mask is None (otherwise it will fall back to the math kernel)
```
Comment (justinchuby) [inline src/transformers/integrations/sdpa_attention.py:34]: Is it required that attention_mask is None? Just wanting to check it the situation has changed

Comment (vasqu) [reply src/transformers/integrations/sdpa_attention.py:34]: Yup, at least on older versions it is! Was only recently relaxed

Comment (vasqu) [reply src/transformers/integrations/sdpa_attention.py:34]: See the description here https://docs.pytorch.org/docs/2.5/generated/torch.nn.functional.scaled_dot_product_attention.html (under gqa --> supports only flash and math --> flash only works without a mask)

We want to avoid the math kernel
